python/ray/train/v2/tests/test_sync_actor.py

```diff
@@ -56,37 +56,32 @@ def test_hang():
     with pytest.raises(BroadcastCollectiveTimeoutError) as excinfo:
         ray.get(remote_tasks)
     # Ensure that "rank 9 workers not reaching barrier" included in the error message
-    assert (
-        "The following ranks are not reaching the collective "
-        "operation: [9]\n" in str(excinfo.value)
-    )
+    assert "The following ranks have not called it: [9]" in str(excinfo.value)
 
 
 def test_world_size_mismatch():
     """The test checks if the workers are blocked and raise an value error
     when the world size is different. The workers should block and raise
     an ValueError.
     """
-    sync_actor = SynchronizationActor.remote()
-    # Test broadcast_from_rank_zero with a world size of 10. But
-    # one of the worker reports a different world size, raise an assertion error.
+    sync_actor = SynchronizationActor.remote(timeout_s=1, warn_interval_s=0.2)
     remote_tasks = []
+
+    # All workers pass use a world size of 10, except for one.
+    # This should raise an assertion error on the offending caller.
     for rank in range(9):
         remote_tasks.append(
             sync_actor.broadcast_from_rank_zero.remote(
                 world_rank=rank, world_size=10, data=f"data-{rank}"
             )
         )
-    # The last worker datas with a different world size
-    remote_tasks.append(
-        sync_actor.broadcast_from_rank_zero.remote(
-            world_rank=9, world_size=11, data="data-9"
-        )
+    # The last worker calls broadcast with a different world size
+    mismatch_task = sync_actor.broadcast_from_rank_zero.remote(
+        world_rank=9, world_size=11, data="data-9"
     )
-    # Ensure that the workers are blocked and getting GetTimeoutError after 5 seconds
-    with pytest.raises(ValueError) as excinfo:
-        ray.get(remote_tasks, timeout=1)
-    assert "ValueError" in str(excinfo.type)
+
+    with pytest.raises(ValueError, match="same world size"):
+        ray.get(mismatch_task, timeout=1)
```

Comment:
nit: Should this timeout be slightly higher than `timeout_s=1` that the `SynchronizationActor` was initialized with to ensure there isn't a race condition between these two timeouts?

Reply:
The timeouts here are not needed actually. I'll remove both of them.

The actor who submits the wrong world size will immediately raise the `ValueError`.